### DISCUSSION TITLE
How can I extend the modifier (make a custom one) for x-on?

### DISCUSSION THREAD

**Discussion by s-patompong** [alpinejs/alpine#2365] [1. Help]
Hi :)

Parts of the project that I'm working on has something like this:
```html
<div
    x-data="{show: false}"
    @mouseenter.self="
      if(isTouchDevice()) return;

      show = true;
   "
>
    <div>Hover me!</div>
    <div x-show="show">I'm a secret, and I'll show up only in a non-touch device!</div>
</div>
```

We try to run the logic inside for the `mouseenter` event only if the device is a non-touch device (PC, for example). As you can see, we have the `isTrouchDevice()` function that can tell us if the device is a touch device or not.

The problem is, we need to add this statement `if(isTouchDevice()) return;` all over the places and we think it might be better if we can add a custom modifier to AlpineJS. Something like this:
```html
<div
    x-data="{show: false}"
    @mouseenter.self.nontouch="
      show = true;
   "
>
    <div>Hover me!</div>
    <div x-show="show">I'm a secret, and I'll show up only in a non-touch device!</div>
</div>
```

Is this possible to create such a custom modifier without customizing the x-on directive source code? I'm just looking for an easy way to extend the modifier :)

Any help is appreciated!

**Comment by SimoTod** [comment]
It's not possible to extends modifiers at the moment.

**Reply by s-patompong** [reply]
I see, thank you for the reply :)

**Comment by Roaann** [comment]
@SimoTod Do you happen to know if it available at this time? I'd like to make a 'double' modifier for the click directive

**Reply by ekwoka** [reply]
Currently no.

It would take quite a refactor to make such things possible.

You could take the existing one and copy paste it and add the changes you want and run it as a directive.
